apps/web/pages/apps/[slug]/[...pages].tsx

```diff
@@ -67,7 +68,11 @@ const AppPage: AppPageType["default"] = function AppPage(props) {
   if (!route || route.notFound) {
     throw new Error("Route can't be undefined");
   }
-  return <route.Component {...componentProps} />;
+  return (
+    <FormProvider {...methods}>
+      <route.Component {...componentProps} />;
```

Comment:
This isn't the right place to use FormProvider. This file is just a Router to render any page for any app. 
FormProvider at the moment is a requirement of the Routing Form only. Ideally we should move it down to app-store/routing-forms

Reply:
Can we not solve it by using the FormProvider in SingleForm.tsx file?

Reply:
Hello @hariombalhara 
We are using it on the page because each tab is rendered by a different component, when switching tabs, the component is unmounted and another is mounted and uses server's data. This component where the provider is being used, does not unmount when switching tabs and this allows us to save the most recent data, not saved yet.

Reply:
We should be able to use this strategy https://nextjs.org/docs/basic-features/layouts#per-page-layouts to preserve state across routes(different components)

We already used it in app <img width="674" alt="Screenshot 2023-04-18 at 7 57 56 PM" src="https://user-images.githubusercontent.com/1780212/232808733-932cad6b-64bf-41b7-a171-ba916e695508.png">